modeshape-jcr/src/main/java/org/modeshape/connector/filesystem/FileSystemConnector.java

```diff
@@ -481,16 +481,30 @@ public boolean hasDocument( String id ) {
         return fileFor(id).exists();
     }
 
+    /**
+     * Hook for processing other node types (not resources or regular files)
+     * @param id
+     * @param file
+     * @param isRoot
+     * @return true if node is recognized and processed; false causes fallback
+     * to processing as a directory.
+     */
+    protected DocumentWriter otherType(String id, File file, boolean isRoot, boolean isResource) {
+        return null;
+    }
+
     @Override
     public Document getDocumentById( String id ) {
         File file = fileFor(id);
         boolean isRoot = isRoot(id);
         // make sure the root folder is never excluded....
-        if (!isRoot && (isExcluded(file) || !file.exists())) return null;
+        if (!isRoot && (isExcluded(file) || !acceptFile(file))) return null;
         boolean isResource = isContentNode(id);
         DocumentWriter writer = null;
```

Comment:
with the above methods, this could be changed to:
`writer = writerFor(...)` and then the check would be simply `if (writer != null)` or else do the regular logic